src/features/search/searchSlice.ts

```diff
@@ -0,0 +1,43 @@
+import { createSelector, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { RootState } from '@common/state/store';
+
+export interface State {
+  searchTerm: string;
+  isFocused: boolean;
+}
+
+const initialState: State = {
+  searchTerm: '',
+  isFocused: false,
+};
+
+export const slice = createSlice({
+  name: 'search',
+  initialState,
+  reducers: {
+    setSearchTerm: (state, action: PayloadAction<string>) => {
+      state.searchTerm = action.payload;
+    },
```

Comment:
What's the requirement that the search term is persisted in the store? Do we need to later?

Generally form values should be kept out of redux imo, [See Dan Abromov comment](https://github.com/reduxjs/redux/issues/1287#issuecomment-175351978)

Reply:
https://goshacmd.com/should-i-put-form-state-into-redux/

Reply:
@kyranjamie do you have a link to the diagram on when to use what state mgmt? Was that in notion?

Reply:
The one I made is in this document https://www.notion.so/hiropbc/State-management-next-steps-9027c10831f54cd8bf40d4566281b0f1

Reply:
@kyranjamie I generally agree with using react state for form inputs, this one is a bit different though as I use the same state for both search bars (homepage and header) and result wrappers. If we use react state, we'll need to call useSearch hook in the parent component and pass props down to search bar and result wrapper (which is not the best IMO) + write it twice in both search bars. Or maybe there's a better idea?